sematic/ui/src/types/Types.tsx

```diff
@@ -318,6 +318,30 @@ function ListTypeView(props: TypeViewProps) {
   );
 }
 
+function UnionValueView(props: ValueViewProps) {
+  const typeRepr = props.typeRepr as AliasTypeRepr;
+
+  if (!typeRepr[2].args) {
+    return ValueView(props);
+  }
+
+  const noneTypeArgs: TypeParamRepr[] = typeRepr[2].args.filter((argsTypeParamRep) => argsTypeParamRep.type[1]==="NoneType");
+  const containsNoneType: boolean = noneTypeArgs.length > 0;
+
+  if (containsNoneType) {
+    const otherTypes: TypeParamRepr[] = typeRepr[2].args.filter((argsTypeParamRep) => argsTypeParamRep.type[1]!=="NoneType");
+
+    //Union Types in Python should not contain more than two types. Therefore, the otherTypes array should be of length 1.
```

Comment:
`Union Types in Python should not contain more than two types ` The said is probably true for the `Optional` type. 

But I am not sure it is always true for a generic Union type. It seems it is allowed to have more than two types.

Eg

```python
Union[Union[int, str], float] == Union[int, str, float]

```

Reply:
Agree with Chance--you can have `Union` of arbitrarily many types in python.

Reply:
Right, I should have mentioned it as Optional Types in the comments. I will correct it in my next revision. I agree that for the actual `Union` type we can have multiple possible types as input.